### PR TITLE
feat: add ioredis support

### PR DESCRIPTION
**Pull request by therightstuff**

### CODE REVIEW

File: src/instrumentations/ioredis/IORedisInstrumentation.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+  getInstrumentation(): IORedisInstrumentation {
+    return new IORedisInstrumentation({
+      dbStatementSerializer: function (cmdName, cmdArgs) {
+        const statement = [cmdName, ...cmdArgs].join(' ');
+        return CommonUtils.payloadStringify(
+          statement,
+          ScrubContext.HTTP_REQUEST_BODY,
+          getSpanAttributeMaxLength()
+        );
```
Comment (saartochner) [inline src/instrumentations/ioredis/IORedisInstrumentation.ts:20]: though this is a cool usage of `dbStatementSerializer`, you can simplify your implementation by moving this code to the response hook

File: src/instrumentations/ioredis/IORedisInstrumentation.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    return 'ioredis';
+  }
+
+  getInstrumentation(): IORedisInstrumentation {
+    return new IORedisInstrumentation({
+      dbStatementSerializer: function (cmdName, cmdArgs) {
+        const statement = [cmdName, ...cmdArgs].join(' ');
+        return CommonUtils.payloadStringify(
+          statement,
+          ScrubContext.HTTP_REQUEST_BODY,
```
Comment (saartochner) [inline src/instrumentations/ioredis/IORedisInstrumentation.ts:18]: hmmmm
`db.statement` is the right attr name, I think

File: src/instrumentations/ioredis/IORedisInstrumentation.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        span: Span,
+        cmdName: string,
+        cmdArgs: (string | Buffer)[],
+        response: unknown
+      ) => {
+        span.setAttribute(
+          `redis.response.body`,
+          CommonUtils.payloadStringify(
+            response?.toString(),
+            ScrubContext.HTTP_RESPONSE_BODY,
```
Comment (saartochner) [inline src/instrumentations/ioredis/IORedisInstrumentation.ts:32]: `redis.response.body`?
why HTTP?

File: src/instrumentations/ioredis/IORedisInstrumentation.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        );
+      },
+      responseHook: (
+        span: Span,
+        cmdName: string,
+        cmdArgs: (string | Buffer)[],
+        response: unknown
+      ) => {
+        span.setAttribute(
+          `redis.response.body`,
```
Comment (saartochner) [inline src/instrumentations/ioredis/IORedisInstrumentation.ts:29]: change to `db.response.body`

Comment (therightstuff) [reply src/instrumentations/ioredis/IORedisInstrumentation.ts:29]: :thumbsup:

File: test/instrumentations/ioredis/ioredisTestUtils.ts [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+  );
+
+  expect(connectionSpans[1]).toMatchObject(
+    getExpectedSpan({
+      name: 'info',
+      resourceAttributes,
+      attributes: {
+        'db.system': 'redis',
+        'db.statement': JSON.stringify('info'),
+        'redis.response.body': expect.any(String),
```
Comment (saartochner) [inline test/instrumentations/ioredis/ioredisTestUtils.ts]: should fail?

File: test/instrumentations/ioredis/ioredisTestUtils.ts [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+  );
+
+  expect(connectionSpans[2]).toMatchObject(
+    getExpectedSpan({
+      name: 'quit',
+      resourceAttributes,
+      attributes: {
+        'db.system': 'redis',
+        'db.statement': JSON.stringify('quit'),
+        'redis.response.body': JSON.stringify('OK'),
```
Comment (saartochner) [inline test/instrumentations/ioredis/ioredisTestUtils.ts]: should fail?

**Review by saartochner — APPROVED** [review]

**Pull request merged**